File: C++/Problems/Problems/problem_1725.cpp
#include "problem_1725.h"

#include <iostream>
#include <bits/ostream.tcc>

bool problem_1725::test() {
    std::vector<std::vector<int>> rectangles { {5, 8},  {3, 9}, {5, 12}, {16, 5}};

    auto expected = 3;

    auto result = countGoodRectangles(rectangles);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_1725::countGoodRectangles(std::vector<std::vector<int>> &rectangles) {
    auto max = 0;
    auto count = 0;
    for (auto r : rectangles)
    {
        auto side = std::min(r[0], r[1]);
        if (side == max)
            count++;
        else if (side > max)
        {
            max = side;
            count = 1;
        }
    }

    return count;
}

File: Python/Problems/Problems/problem1725.py
from .problem import Problem

class Problem1725(Problem):
    def test(self):
        rectangles = [[5, 8], [3, 9], [5, 12], [16, 5]]

        expected = 3

        result = self.countGoodRectangles(rectangles)

        print(result)

        return result == expected

    def countGoodRectangles(self, rectangles):
        max = 0
        count = 0
        for r in rectangles:
            side = min(r[0], r[1])
            if side == max:
                count += 1
            elif side > max:
                max = side
                count = 1

        return count